pkg/networkValidator/networkValidator.go

```diff
@@ -21,7 +22,7 @@ func (nv *NetworkValidator) SetOfflineMode(offlineMode string) {
 }
 
 func (nv *NetworkValidator) IdentifyNetworkError(errStr string) error {
-	if errStr == "network error" {
+	if errStr == "network error" || strings.Contains(errStr, "connection refused") || strings.Contains(errStr, "no such host") {
 		if nv.offlineMode == "fail" {
```

Comment:
We can't really cover 100% of network errors due to network components react to no internet / backend is down. ATM we focus on adding more possible errors